plugins/woocommerce/includes/class-wc-ajax.php

```diff
@@ -3030,8 +3030,14 @@ public static function shipping_zone_add_method() {
 
 		$zone_id     = wc_clean( wp_unslash( $_POST['zone_id'] ) );
 		$zone        = new WC_Shipping_Zone( $zone_id );
+		do_action( 'woocommerce_update_shipping_setting', array( 'id' => 'zone_method' ) );
 		$instance_id = $zone->add_shipping_method( wc_clean( wp_unslash( $_POST['method_id'] ) ) );
 
+
+		global $current_tab;
+		$current_tab = 'shipping';
```

Comment:
Oh globals. I guess that since the current implementation depends on this, it is okay. 😄